CharlieBackend.Business/Services/MentorService.cs

```diff
@@ -65,12 +65,15 @@ public async Task<MentorModel> CreateMentorAsync(CreateMentorModel mentorModel)
                     }
 
                     await _unitOfWork.CommitAsync();
-                    await _credentialsSender.SendCredentialsAsync(account.Email, generatedPassword);
 
-                    transaction.Commit();
+                    if (await _credentialsSender.SendCredentialsAsync(account.Email, generatedPassword))
+                    {
+                        transaction.Commit();
 
-                    return mentor.ToMentorModel();
+                        return mentor.ToMentorModel();
+                    }
```

Comment:
Please allow Mentor registration in case of even EMAIL is not sent. If email is not sent, should not stop account registration.

Reply:
Changes has been added.